x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolverTests.java

```diff
@@ -618,6 +618,31 @@ public void testResolveWildcardsMinusExpandWilcardsOpenAndClosed() {
         assertThat(request.indices(), arrayContainingInAnyOrder(replacedIndices));
     }
 
+    public void testResolveWildcardsNoExpand() {
+        SearchRequest request = new SearchRequest("*", "-foofoo*");
+        // no wildcard expand and no ignore unavailable
+        request.indicesOptions(IndicesOptions.fromOptions(false, randomBoolean(), false, false));
+        ResolvedIndices indices = resolveIndices(request, buildAuthorizedIndices(user, SearchAction.NAME));
+        String[] replacedIndices = new String[] { "*", "-foofoo*" };
+        assertThat(indices.getLocal(), hasSize(replacedIndices.length));
+        assertThat(request.indices().length, equalTo(replacedIndices.length));
+        assertThat(indices.getLocal(), hasItems(replacedIndices));
+        assertThat(request.indices(), arrayContainingInAnyOrder(replacedIndices));
```

Comment:
Nit:
```suggestion
        assertThat(indices.getLocal(), containsInAnyOrder(replacedIndices));
        assertThat(request.indices(), arrayContainingInAnyOrder(replacedIndices));
```

since containsInAnyOrder mandates length to be equal.